rllib/algorithms/impala/impala.py

```diff
@@ -141,6 +147,14 @@ def __init__(self, algo_class=None):
         # Deprecated value.
         self.num_data_loader_buffers = DEPRECATED_VALUE
 
+    @override(AlgorithmConfig)
+    def rollouts(self, *args, **kwargs) -> "AlgorithmConfig":
+        super().rollouts(*args, **kwargs)
+        self.learner_hps.rollout_frag_or_episode_len = (
```

Comment:
need to provide the rollout frag length in order to do vtrace. But at the same time what if someone does complete episodes mode?

Reply:
thats fair. I thought that would be a solution as well but that it would be invalid because we don't like modifying our configs inside of the validate fn. Then again, we aren't changing the hparams at all, so I guess there shouldn't be a problem.

Reply:
you should raise an error if it's already set or something and then set it. This way it's explicit and there is not magic behind the scene.